insonmnia/worker/storagequota/btrfs/btrfs_api.go

```diff
@@ -0,0 +1,156 @@
+// +build linux
+
+package btrfs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	log "github.com/noxiouz/zapctx/ctxlog"
+	"go.uber.org/zap"
+)
+
+type API interface {
+	QuotaEnable(ctx context.Context, path string) error
+	QuotaExists(ctx context.Context, qgroupID string, path string) (bool, error)
+	QuotaCreate(ctx context.Context, qgroupID string, path string) error
+	QuotaDestroy(ctx context.Context, qgroupID string, path string) error
+	QuotaLimit(ctx context.Context, sizeInBytes uint64, qgroupID string, path string) error
+	QuotaAssign(ctx context.Context, src string, dst string, path string) error
+	QuotaRemove(ctx context.Context, src string, dst string, path string) error
+	GetQuotaID(ctx context.Context, path string) (string, error)
+}
+
+var _ API = btrfsCLI{}
+
+func NewAPI() (API, error) {
+	if _, err := exec.LookPath("btrfs"); err != nil {
+		return nil, fmt.Errorf("btrfs executable is not found. Install btrfs-progs")
+	}
+	return btrfsCLI{}, nil
+}
+
+type btrfsCLI struct{}
+
+func (btrfsCLI) QuotaEnable(ctx context.Context, path string) error {
+	output, err := exec.CommandContext(ctx, "btrfs", "quota", "enable", path).Output()
+	if err != nil {
+		log.G(ctx).Error("failed to create to enable btrfs quota", zap.String("path", path), zap.Error(err), zap.ByteString("output", output))
+		return err
+	}
+	return nil
+}
+
+func (btrfsCLI) QuotaExists(ctx context.Context, qgroupID string, path string) (bool, error) {
+	// TODO: add implementation
+	output, err := exec.CommandContext(ctx, "btrfs", "qgroup", "show", path).Output()
+	if err != nil {
+		log.G(ctx).Error("failet to lookup quota", zap.String("path", path), zap.String("quota", qgroupID), zap.Error(err))
+	}
+
+	return lookupQuotaInShowOutpout(output, qgroupID)
+}
+
+func lookupQuotaInShowOutpout(output []byte, qgroupID string) (bool, error) {
+	qgroupIDBytes := []byte(qgroupID)
+	r := bytes.NewReader(output)
+	scanner := bufio.NewScanner(r)
+	foundHeader := false
+	endOfHeaderToken := []byte("--------") // next line after qgroupid
```

Comment:
shiiiet

Reply:
Look at tests :) do not know how to detect it